packages/stargate/src/aminotypes.ts

```diff
@@ -27,11 +27,16 @@ export class AminoTypes {
   private readonly register: Record<string, string>;
 
   public constructor(additions: Record<string, string> = {}) {
-    this.register = { ...defaultTypes, ...additions };
+    const additionalAminoTypes = Object.values(additions);
+    const filteredDefaultTypes = Object.entries(defaultTypes).reduce(
+      (acc, [key, value]) => (additionalAminoTypes.includes(value) ? acc : { ...acc, [key]: value }),
+      {},
+    );
```

Comment:
Oh, why is this needed?

Reply:
Otherwise, it might be many-to-one and `.find` will find the wrong one.

Reply:
(I.e. the third test for `fromAmino` will fail.

Reply:
Great, thanks